pkg/provider/cloud/kubevirt/client.go

```diff
@@ -48,6 +60,7 @@ type ClientOptions struct {
 
 func newClient(kubeconfig string, opts ClientOptions) (*Client, error) {
 	var client ctrlruntimeclient.Client
+	opts.ControllerRuntimeOptions.Scheme = scheme
```

Comment:
is it a good idea to override this unconditionally? That way you cannot pass a custom scheme anymore, even if you wanted. It would be nice if `opts.ControllerRuntimeOptions.Scheme` was respected if it's not `nil`, and otherwise we use the new scheme added, no?

Reply:
I am not convinced that we want to pass a custom scheme for KubeVirt client.

> That way you cannot pass a custom scheme anymore

That's the goal. If you want to do that just use another controller runtime client instance.
Having a strict control over schema makes this piece of code a KubeVirt client not another generic controller-runtime client.

I want to avoid situation where people using this client override (or even merge) the schema.
It should be related to our KubeVirt features as much as possible.

That's my reasoning.

Reply:
However, I could be more explicit about that and potentially return an error when somebody tries to pass schema in the options.

Reply:
Yeah I think throwing an error would be nice. I agree it's probably not a good idea to use a different scheme, but just dropping it silently could lead to confusion later on.

Reply:
I plan to improve in the near future, for now I would vote to leave it as it is.
I will get rid of the client duplication in the dashboard so at that time i will add necessary checks.

Reply:
alright, makes sense.